src/targetPacks/assemble.targets

```diff
@@ -1,19 +1,90 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <ItemGroup>
-    <TargetingPackSrc Include="$(MSBuildProjectDirectory)/**/*.il" />
+      <!--
+        Files that may need the 'constraint' token fixed before they can be assembled. They were
+        decompiled with a 3.1 ildasm not compatible with 5.0 ilasm. In some cases the fix is not
+        necessary for a given IL filename in a certain TFM, but is necessary for a file with the
+        same name in a different TFM. The patch doesn't take a particularly long time, so prefer
+        keeping this list simple.
+      -->
+      <TargetingPackSrcRequiringConstraintPatch
+        Include="
+          $(MSBuildProjectDirectory)\**\Microsoft.AspNetCore.Components.il;
+          $(MSBuildProjectDirectory)\**\Microsoft.AspNetCore.Mvc.Cors.il;
+          $(MSBuildProjectDirectory)\**\Microsoft.AspNetCore.Routing.il;
+          $(MSBuildProjectDirectory)\**\netstandard.il;
+          $(MSBuildProjectDirectory)\**\presentationframework.*.il;
+          $(MSBuildProjectDirectory)\**\PresentationFramework.*.il;
+          $(MSBuildProjectDirectory)\**\PresentationFramework.il;
+          $(MSBuildProjectDirectory)\**\System.Activities.Presentation.il;
+          $(MSBuildProjectDirectory)\**\System.ComponentModel.Composition.il;
+          $(MSBuildProjectDirectory)\**\System.Data.Common.il;
+          $(MSBuildProjectDirectory)\**\System.Data.Entity.il;
+          $(MSBuildProjectDirectory)\**\System.Data.il;
+          $(MSBuildProjectDirectory)\**\System.Design.il;
+          $(MSBuildProjectDirectory)\**\System.Reflection.Metadata.il;
+          $(MSBuildProjectDirectory)\**\System.Web.il;
+          $(MSBuildProjectDirectory)\**\System.Xml.il;
+          $(MSBuildProjectDirectory)\**\System.XML.il;
+          $(MSBuildProjectDirectory)\**\WindowsFormsIntegration.il" />
+
+      <!-- Find all IL. Leave out IL that needs to be patched before being built. -->
+      <TargetingPackSrc
+        Include="$(MSBuildProjectDirectory)\**\*.il"
+        Exclude="@(TargetingPackSrcRequiringConstraintPatch)" />
   </ItemGroup>
-  <Target Name="AssembleTargetPack" Condition="'@(TargetingPackSrc)' != '' and '$(SkipTargetingPacks)' != 'true' ">
+
+  <Target Name="GetProjectSrc"
+          BeforeTargets="PatchIl">
+    <PropertyGroup>
+      <PatchedILSrcTemp>$(RepoRoot)artifacts/patchedILSrc/</PatchedILSrcTemp>
+    </PropertyGroup>
+    <ItemGroup>
+      <TargetingPackSrc
+        ILSrcFile="%(FullPath)"
+        RelativeOutputAssemblyFile="%(RecursiveDir)%(Filename).dll" />
+      <TargetingPackSrcRequiringConstraintPatch
+        ILSrcFile="$(PatchedILSrcTemp)$(MSBuildProjectName)/%(RecursiveDir)%(Filename).patched%(Extension)"
+        RelativeOutputAssemblyFile="%(RecursiveDir)%(Filename).dll"
+        DestinationFile="$(PatchedILSrcTemp)$(MSBuildProjectName)/%(RecursiveDir)%(Filename).patched%(Extension)" />
+      <AllSrc
```

Comment:
I'd suggest pulling descriptiveness up if the separate item lists are really necessary (rather than just putting the items into `TargetingPackSrc` like `master`):

`TargetingPackSrc`, `TargetingPackSrcRequiringConstraintPatch`, `AllSrc` ->
`TargetingPackSrcNotRequiringConstraintPatch`, `TargetingPackSrcRequiringConstraintPatch`, `TargetingPackSrc`

Reply:
I changed to use these names.